resilience4j-timelimiter/src/test/java/io/github/resilience4j/timelimiter/TimeLimiterTest.java

```diff
@@ -8,41 +8,66 @@
 import java.util.function.Supplier;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.assertj.core.api.BDDAssertions.then;
-import static org.mockito.Mockito.*;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.then;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
 
 public class TimeLimiterTest {
 
     @Test
     public void shouldReturnCorrectTimeoutDuration() {
         Duration timeoutDuration = Duration.ofSeconds(1);
         TimeLimiter timeLimiter = TimeLimiter.of(timeoutDuration);
-        then(timeLimiter).isNotNull();
-        then(timeLimiter.getTimeLimiterConfig().getTimeoutDuration()).isEqualTo(timeoutDuration);
+        assertThat(timeLimiter).isNotNull();
+        assertThat(timeLimiter.getTimeLimiterConfig().getTimeoutDuration()).isEqualTo(timeoutDuration);
     }
 
     @Test
-    public void shouldThrowTimeoutExceptionAndInvokeCancel() throws InterruptedException, ExecutionException, TimeoutException {
+    public void shouldThrowTimeoutExceptionAndInvokeCancel() throws Exception {
         Duration timeoutDuration = Duration.ofSeconds(1);
         TimeLimiter timeLimiter = TimeLimiter.of(timeoutDuration);
 
         @SuppressWarnings("unchecked")
         Future<Integer> mockFuture = (Future<Integer>) mock(Future.class);
 
         Supplier<Future<Integer>> supplier = () -> mockFuture;
-        when(mockFuture.get(timeoutDuration.toMillis(), TimeUnit.MILLISECONDS)).thenThrow(new TimeoutException());
+        given(mockFuture.get(timeoutDuration.toMillis(), TimeUnit.MILLISECONDS)).willThrow(new TimeoutException());
 
         Callable<Integer> decorated = TimeLimiter.decorateFutureSupplier(timeLimiter, supplier);
         Try<Integer> decoratedResult = Try.ofCallable(decorated);
 
-        then(decoratedResult.isFailure()).isTrue();
-        then(decoratedResult.getCause()).isInstanceOf(TimeoutException.class);
+        assertThat(decoratedResult.isFailure()).isTrue();
+        assertThat(decoratedResult.getCause()).isInstanceOf(TimeoutException.class);
 
-        verify(mockFuture).cancel(true);
+        then(mockFuture).should().cancel(true);
     }
 
     @Test
-    public void shouldThrowTimeoutExceptionAndNotInvokeCancel() throws InterruptedException, ExecutionException, TimeoutException {
+    public void shouldThrowTimeoutExceptionWithCompletionStage() throws Exception {
+        Duration timeoutDuration = Duration.ofSeconds(1);
```

Comment:
Can we somehow speed up this test? Maybe reduce the timeoutduration?

Reply:
Mocking is most fast way. But for credible test, I'll reduce duration.